yt_dlp/extractor/common.py

```diff
@@ -3687,6 +3687,26 @@ def _extract_chapters_from_description(self, description, duration):
             start_function=lambda x: parse_duration(x[1]), title_function=lambda x: x[0],
             duration=duration, strict=False)
 
+    def _extract_heatmap_helper(self, heatmap_list, start_function, end_function, value_function, strict=True):
+        heatmap_list = [{
+            'start_time': start_function(point),
+            'end_time': end_function(point),
+            'value': value_function(point),
+        } for point in heatmap_list or []]
+        if strict:
+            warn = self.report_warning
+        else:
+            warn = self.write_debug
+            heatmap_list.sort(key=lambda p: p['start_time'] or 0)
```

Comment:
Why are we splitting the logic into `strict`? Is there a case where we wouldn't want to sort? This would also simplify the for logic below, since we can assume it will always be sorted?